### PR TITLE
Introduce signature_separate config option

### PR DESCRIPTION
**Pull request by mrvdb**
When true, the signature part of the message will be separated from
the message by '-- \n'

Related issue #371

### CODE REVIEW

File: src/account_manager.cc [hunk 1]
```diff
@@ -39,6 +39,7 @@ namespace Astroid {
 
       a->save_drafts_to = kv.second.get<string> ("save_drafts_to");
 
+      a->signature_separate = kv.second.get<bool> ("signature_separate");
```
Comment (gauteh) [inline src/account_manager.cc]: on second thought this should probably be "signature_separator" ?

**Comment by gauteh** [comment]
Looks good: we need to support seamless config-update on this one, probably have to bump config version as well. Have a look here for earlier examples: https://github.com/astroidmail/astroid/blob/master/src/config.cc#L431

We can also probably start to remove the old migrations - I don't think we need to support config files that are more than 2 years old.

File: src/account_manager.cc (hunk 1 above)
Comment (mrvdb) [inline src/account_manager.cc]: I had it like that first (as a string config option) which specified the separator, if it is a boolean 'signature_separate' is better, no?

Comment (gauteh) [inline src/account_manager.cc]: ok! sounds good.

**Comment by gauteh** [comment]
Also, you have to add the default setting to the example config in config.cc.

**Comment by mrvdb** [comment]
Something like this you mean?

```
diff --git a/src/config.cc b/src/config.cc
index 782dac3..71401fd 100644
--- a/src/config.cc
+++ b/src/config.cc
@@ -146,6 +146,7 @@ namespace Astroid {
       default_config.put ("accounts.charlie.save_drafts_to",
           "/home/root/Mail/drafts/");
 
+      default_config.put ("accounts.charlie.signature_separate", false);
       default_config.put ("accounts.charlie.signature_file", "");
       default_config.put ("accounts.charlie.signature_default_on", true);
       default_config.put ("accounts.charlie.signature_attach", false);
@@ -439,6 +440,18 @@ namespace Astroid {
             config.put (key.c_str (), false);
           }
         }
+
+       if (version < 8) {
+         try {
+           
+           ustring ss = kv.second.get<bool> ("signature_separate");
+           
+         } catch (const boost::property_tree::ptree_bad_path &ex) {
+           
+           ustring key = ustring::compose ("accounts.%1.signature_separate", kv.first);
+           config.put (key.c_str (), false);
+          }
+       }
       }
 
       changed = true;
```

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 14:53:
> Something like this you mean?
> 
> ```
> +           config.put (key.c_str (), false);
> ```

Should this be the same as the default? This will 
cause a bit of a hassle for users, but we'll have to write it in the 
History.txt / release announcement. Also, it's not the end of the world 
if there is a few double '--'s.

Please test if upgrading the config works without hassle (the option 
should be added without error if it is not present).

Config version is set in config.hh if I remember correctly.

- gaute

**Comment by gauteh** [comment]
Otherwise the config stuff looks good :)

**Comment by mrvdb** [comment]
I think i've addressed all hints and tips (thanks!) and from my point of view, this can be merged.

**Comment by gauteh** [comment]
Thanks, merged! I fixed up the commit messages and tweaked the History a bit.

**Pull request closed without merging**